hazelcast/src/main/java/com/hazelcast/map/impl/PartitionContainer.java

```diff
@@ -225,4 +239,29 @@ public void setLastCleanupTimeCopy(long lastCleanupTimeCopy) {
         this.lastCleanupTimeCopy = lastCleanupTimeCopy;
     }
 
+    // -------------------------------------------------------------------
+    // WATCH OUT: never use directly! use MapContainer.getIndex() instead
```

Comment:
can you please also add the reason for not using directly? 
Also is it possible to throw some exception in case of misuse?